Intro/README.md

```diff
@@ -0,0 +1,455 @@
+# Introduction to TPMs
+
+Trusted Platform Modules (TPMs) are a large and complex topic, made all
+the more difficult to explain by the intricate relationships between the
+relevant concepts.  This is an attempt at a simple explanation --
+simpler than reading hundreds of pages of documents, but then too, too
+light on detail to be immediately useful.
+
+So what is a TPM?  Well, it's a cryptographic co-processor with special
+features to enable "root of trust measurement" (RTM), remote attestation
+of system state, unlocking of local resources that are kept encrypted
+(e.g., filesystems), and more.  A TPM can do those things, and it can do
+it with rich authentication and authorization policies.
+
+Typically a TPM is a hardware module, a chip, though there are firmware,
+virtual, and simulated TPMs as well, all implemented in software.
+
+To simplify things we'll consider only TPM 2.0.  Also to simplify things
+we'll ignore algorithm agility.
+
+Other parts of this [tutorial](README.md) may cover specific concepts in
+much more detail.
+
+# Core Concepts
+
+Some core concepts in the world of TPMs
+
+ - cryptography:
+    - digest functions
+    - hash extension
+    - HMAC
+    - symmetric encryption
+    - asymmetric encryption and/or signature algorithms
+ - platform configuration registers (PCRs)
+ - cryptographic object naming
+ - immutability of object public areas
+ - key hierarchies
+ - key wrapping
+ - limited resources
+    - tickets
+    - resource management
+ - sessions
+ - authorization
+    - restricted cryptographic keys
+    - policies
+ - other object types
+    - non-volatile (NV) indexes
+ - attestation
+
+We'll assume reader familiarity with cryptography so we need not explain
+it.
+
+Authorization is the most important aspect of a TPM, since that's
+ultimately what it exists for: to authorize a system or application to
+perform certain duties when all the desired conditions allow for it.
+
+TPMs have a very rich set of options for authorization.  It's not just
+[policies](#Policies), but also cryptographic object names used with
+restricted keys to allow access only to applications that also have
+other access.
+
+Where to start?  Let's start with hash extension, which may be the only
+trivial concept in the world of TPMs!
+
+## Important Notes
+
+ - Chained implications.
+
+   It is important to note the TPM ecosystem's critical dependence on
+   TPMs performing authorization as specified.  A lot can be
```

Comment:
"critical dependence on TPMs performing authorization as specified": Is this any different than "we expect computers to compute properly"? I'm not sure why I wouldn't expect the TPM to behave correctly. Are you getting at whether the TPM is trustworthy? And whether that trustworthiness is different than the rest of the computer hardware? If so you might need to explain that a bit more.

Reply:
Ah, I've something very specific in mind here, which is restricted keys, specifically `TPM2_ActivateCredential()` and the other side of it, `TPM2_MakeCredential()`, and basically you only want to send a "credential" to a TPM you know is legitimate.  Except of course that the TPM might be a virtual TPM, implemented in software in a hypervisor, and then what?  Well, the idea is that you authenticate the TPM's EKpub/EKcert before you call `TPM2_MakeCrednetial()` to send it a credential.

OK, so maybe this is too soon to hint at that and I should just remove that text.

Reply:
Follow-up from our TPM.dev weekly call this week: @kgoldman checked in the spec and said that a restricted key is a TPM key that can be used only on TPM-generated data. For example, Attestation Keys can only sign TPM-generated data. However, the restricted key does not have a fixedTPM attribute (which is what I thought).

Reply:
> 
> Follow-up from our TPM.dev weekly call this week: @kgoldman checked in the spec and said that a restricted key is a TPM key that can be used only on TPM-generated data. 

It can sign any message that the TPM has hashed.  An externally generated message cannot start with the magic number.